Windows only: Workflow step 'setDevVersion' is failing on git repo if Maven module file path name contains upper case characters

## Issue
If a Maven module is located in a folder named with uppercase characters, then
unleash git SCM provider is failing within workflow step setDevVersion. 
This issue occurs on Windows OS runtimes only.

## Environment
- OS: Windows
- SCM: Git

## Root cause
Class `FileToRelativePath.java` line 30-42 converting all file path characters to lower case in case of Windows OS:
https://github.com/mavenplugins/unleash-maven-plugin/blob/353a9d0b5cf412a07b7c52fdeedb902a15b1848d/scm-provider-api/src/main/java/com/itemis/maven/plugins/unleash/scm/utils/FileToRelativePath.java#L30-L42

## Log file (extract)
```
...
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for maven-project-artifactId 8.6.0.1:
[INFO] 
[INFO] maven-project-artifactId .......................... SUCCESS [  9.827 s]
[INFO] maven-project-artifactId-integration-shared ....... SUCCESS [  6.498 s]
[INFO] maven-project-artifactId-core ..................... SUCCESS [05:50 min]
[INFO] maven-project-artifactId-eihub .................... SUCCESS [  1.643 s]
[INFO] maven-project-artifactId-eihub-cpl-process ........ SUCCESS [  4.571 s]
[INFO] maven-project-artifactId-installer ................ SUCCESS [ 37.997 s]
[INFO] maven-project-artifactId-masterdata-installer ..... SUCCESS [  0.462 s]
[INFO] maven-project-artifactId-masterdata-VIP1 .......... SUCCESS [03:27 min]
[INFO] maven-project-artifactId-masterdata-ZWO1-LACC ..... SUCCESS [03:14 min]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  13:34 min
[INFO] Finished at: 2025-08-01T18:33:42+02:00
[INFO] ------------------------------------------------------------------------
[INFO] [jenkins-event-spy] Generated D:\Jenkins\workspace\PROJECT_master@tmp\withMaven3ee0a158\maven-spy-20250801-182008-1833943597216557140951.log
Picked up JAVA_TOOL_OPTIONS: -Dmaven.ext.class.path="D:\Jenkins\workspace\PROJECT_master@tmp\withMaven3ee0a158\pipeline-maven-spy.jar" -Dorg.jenkinsci.plugins.pipeline.maven.reportsFolder="D:\Jenkins\workspace\PROJECT_master@tmp\withMaven3ee0a158" 
[INFO] 18:33:56,356  Removing io.github.mavenplugins:artifact-spy-plugin:maven-plugin:1.1.0 from build configuration.
[INFO] 18:33:56,394  Checking remote SCM repository for changes. Initial revision was f0a3c120e0dae925118367b383a4637f01902021
[INFO] 18:33:56,394  Tagging local and remote SCM repositories.
[INFO] 18:33:56,421  Git - Searching for Git tag 'PROJECT_RELEASE_V08_006_B01'
[INFO] 18:33:57,747  Git - Tagging local repository with 'PROJECT_RELEASE_V08_006_B01'
[INFO] 18:33:57,748  Git - Committing local changes.
[INFO] 18:34:18,640  Git - Commit finished successfully. New revision is: 42dd9744adab92f76693351d34b01b3ea8bd2e83
[INFO] 18:34:25,611  Git - remote: --------------------------------------------------------------------
[INFO] 18:34:25,611  Git - remote: GL-HOOK-INFO: Hook pre-receive.rb version:        v1.02.00.001
[INFO] 18:34:25,611  Git - remote: GL-HOOK-INFO: rev_old:                            0000000000000000000000000000000000000000
[INFO] 18:34:25,611  Git - remote: GL-HOOK-INFO: rev_new:                            fd47f948e25856011ccfff802d9c481282e67300
[INFO] 18:34:25,611  Git - remote: GL-HOOK-INFO: ref:                                refs/tags/PROJECT_RELEASE_V08_006_B01
[INFO] 18:34:25,611  Git - remote: GL-HOOK-INFO: Using config from default repository.
[INFO] 18:34:25,611  Git - remote: GL-HOOK-INFO: Project is a fork:                  No
[INFO] 18:34:25,611  Git - remote: GL-HOOK-INFO: Total number of commits to check:   1
[INFO] 18:34:25,611  Git - remote: GL-HOOK-INFO: Elapsed time to collect commit(s):  0.022s
[INFO] 18:34:25,611  Git - remote: GL-HOOK-INFO: WebService URL:                     https://pre-receive-hook.git.host.com
[INFO] 18:34:25,611  Git - remote: GL-HOOK-INFO: WebService readTimeout:             299s
[INFO] 18:34:25,611  Git - remote: GL-HOOK-INFO: WebService maxCommitsPerRequest:    100
[INFO] 18:34:25,611  Git - remote: GL-HOOK-INFO: Commits checked via WebService:     1
[INFO] 18:34:25,611  Git - remote: GL-HOOK-INFO: Elapsed time for all commit checks: 1.607s
[INFO] 18:34:26,600  Git - Tag creation finished successfully. New revision is: f0a3c120e0dae925118367b383a4637f01902021
[INFO] 18:34:26,600  Detecting all release artifacts that have been produced during the release build for later installation and ****.
[INFO] 18:34:26,649  Preparing project modules for next development cycle.
[INFO] 18:34:27,696  Git - Committing local changes.
[ERROR] An exception was caught while processing the workflow step with id 'setDevVersion'.
org.eclipse.jgit.api.errors.JGitInternalException: Entry not found by path: maven-project-artifactId-masterdata-installer/vip1/pom.xml
    at org.eclipse.jgit.api.CommitCommand.createTemporaryIndex (CommitCommand.java:508)
    at org.eclipse.jgit.api.CommitCommand.call (CommitCommand.java:226)
    at com.itemis.maven.plugins.unleash.scm.providers.ScmProviderGit.commit (ScmProviderGit.java:405)
    at com.itemis.maven.plugins.unleash.util.DevVersionUtil.commitChanges (DevVersionUtil.java:108)
    at com.itemis.maven.plugins.unleash.steps.actions.SetNextDevVersion.execute (SetNextDevVersion.java:69)
    at com.itemis.maven.plugins.cdi.internal.util.workflow.WorkflowExecutor.executeSequentialWorkflowStep (WorkflowExecutor.java:160)
    at com.itemis.maven.plugins.cdi.internal.util.workflow.WorkflowExecutor.execute (WorkflowExecutor.java:119)
    at com.itemis.maven.plugins.cdi.AbstractCDIMojo.execute (AbstractCDIMojo.java:209)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:299)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:193)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:106)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:963)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:296)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:199)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[INFO] Rolling back after execution errors - please find the error messages and stack traces above.
[INFO] 18:34:28,058  Rollback of all pom changes necessary for setting of the development version as well as reverting any made SCM commits.
[INFO] 18:34:28,079  Rollback of SCM tag creation and POM modifications due to a processing exception.
[INFO] 18:34:28,080  Git - Searching for Git tag 'PROJECT_RELEASE_V08_006_B01'
[INFO] 18:34:29,173  Git - Deleting Git tag
[INFO] 18:34:29,174  Git - Searching for Git tag 'PROJECT_RELEASE_V08_006_B01'
[ERROR] An exception was caught while rolling back the workflow step with id 'tagScm'. Proceeding with the rollback of the next steps.
java.lang.reflect.InvocationTargetException
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at com.itemis.maven.plugins.cdi.internal.util.workflow.WorkflowExecutor.rollback (WorkflowExecutor.java:272)
    at com.itemis.maven.plugins.cdi.internal.util.workflow.WorkflowExecutor.rollback (WorkflowExecutor.java:251)
    at com.itemis.maven.plugins.cdi.internal.util.workflow.WorkflowExecutor.executeSequentialWorkflowStep (WorkflowExecutor.java:164)
    at com.itemis.maven.plugins.cdi.internal.util.workflow.WorkflowExecutor.execute (WorkflowExecutor.java:119)
    at com.itemis.maven.plugins.cdi.AbstractCDIMojo.execute (AbstractCDIMojo.java:209)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:299)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:193)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:106)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:963)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:296)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:199)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: com.itemis.maven.plugins.unleash.scm.ScmException: An error occurred during the deletion of tag 'PROJECT_RELEASE_V08_006_B01' from remote 'origin[https://git.host.com/project.git]'
    at com.itemis.maven.plugins.unleash.scm.providers.ScmProviderGit.callPush (ScmProviderGit.java:519)
    at com.itemis.maven.plugins.unleash.scm.providers.ScmProviderGit.deleteTag (ScmProviderGit.java:792)
    at com.itemis.maven.plugins.unleash.steps.actions.TagScm.rollback (TagScm.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at com.itemis.maven.plugins.cdi.internal.util.workflow.WorkflowExecutor.rollback (WorkflowExecutor.java:272)
    at com.itemis.maven.plugins.cdi.internal.util.workflow.WorkflowExecutor.rollback (WorkflowExecutor.java:251)
    at com.itemis.maven.plugins.cdi.internal.util.workflow.WorkflowExecutor.executeSequentialWorkflowStep (WorkflowExecutor.java:164)
    at com.itemis.maven.plugins.cdi.internal.util.workflow.WorkflowExecutor.execute (WorkflowExecutor.java:119)
    at com.itemis.maven.plugins.cdi.AbstractCDIMojo.execute (AbstractCDIMojo.java:209)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:299)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:193)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:106)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:963)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:296)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:199)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.eclipse.jgit.api.errors.TransportException: Missing unknown 0000000000000000000000000000000000000000
    at org.eclipse.jgit.api.PushCommand.call (PushCommand.java:147)
    at com.itemis.maven.plugins.unleash.scm.providers.ScmProviderGit.callPush (ScmProviderGit.java:485)
    at com.itemis.maven.plugins.unleash.scm.providers.ScmProviderGit.deleteTag (ScmProviderGit.java:792)
    at com.itemis.maven.plugins.unleash.steps.actions.TagScm.rollback (TagScm.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at com.itemis.maven.plugins.cdi.internal.util.workflow.WorkflowExecutor.rollback (WorkflowExecutor.java:272)
    at com.itemis.maven.plugins.cdi.internal.util.workflow.WorkflowExecutor.rollback (WorkflowExecutor.java:251)
    at com.itemis.maven.plugins.cdi.internal.util.workflow.WorkflowExecutor.executeSequentialWorkflowStep (WorkflowExecutor.java:164)
    at com.itemis.maven.plugins.cdi.internal.util.workflow.WorkflowExecutor.execute (WorkflowExecutor.java:119)
    at com.itemis.maven.plugins.cdi.AbstractCDIMojo.execute (AbstractCDIMojo.java:209)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:299)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:193)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:106)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:963)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:296)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:199)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.eclipse.jgit.errors.TransportException: Missing unknown 0000000000000000000000000000000000000000
    at org.eclipse.jgit.transport.Transport.push (Transport.java:1379)
    at org.eclipse.jgit.api.PushCommand.call (PushCommand.java:137)
    at com.itemis.maven.plugins.unleash.scm.providers.ScmProviderGit.callPush (ScmProviderGit.java:485)
    at com.itemis.maven.plugins.unleash.scm.providers.ScmProviderGit.deleteTag (ScmProviderGit.java:792)
    at com.itemis.maven.plugins.unleash.steps.actions.TagScm.rollback (TagScm.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at com.itemis.maven.plugins.cdi.internal.util.workflow.WorkflowExecutor.rollback (WorkflowExecutor.java:272)
    at com.itemis.maven.plugins.cdi.internal.util.workflow.WorkflowExecutor.rollback (WorkflowExecutor.java:251)
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -82,7 +82,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [3.3.1]
 <!-- !!! Align version in badge URLs as well !!! -->
-[![3.3.1 Badge](https://img.shields.io/nexus/r/io.github.mavenplugins/unleash-maven-plugin?server=https://s01.oss.sonatype.org&label=Maven%20Central&queryOpt=:v=3.3.1)](https://central.sonatype.com/artifact/io.github.mavenplugins/unleash-maven-plugin/3.3.1)
+[![3.3.1 Badge](https://img.shields.io/maven-central/v/io.github.mavenplugins/unleash-maven-plugin?label=Maven%20Central&filter=3.3.1)](https://central.sonatype.com/artifact/io.github.mavenplugins/unleash-maven-plugin/3.3.1)
 
 ### Summary
 - Fix issue on Windows OS failing if Maven module relative path names contain upper case characters - #28, #29
@@ -99,7 +99,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [3.3.0]
 <!-- !!! Align version in badge URLs as well !!! -->
-[![3.3.0 Badge](https://img.shields.io/nexus/r/io.github.mavenplugins/unleash-maven-plugin?server=https://s01.oss.sonatype.org&label=Maven%20Central&queryOpt=:v=3.3.0)](https://central.sonatype.com/artifact/io.github.mavenplugins/unleash-maven-plugin/3.3.0)
+[![3.3.0 Badge](https://img.shields.io/maven-central/v/io.github.mavenplugins/unleash-maven-plugin?label=Maven%20Central&filter=3.3.0)](https://central.sonatype.com/artifact/io.github.mavenplugins/unleash-maven-plugin/3.3.0)
 
 ### Summary
 - Fix overriding POM artifact with a file built in a POM type project. Root caused by [artifact-spy-plugin #4](https://github.com/mavenplugins/artifact-spy-plugin/issues/4) - #23
@@ -155,7 +155,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [3.2.1]
 <!-- !!! Align version in badge URLs as well !!! -->
-[![3.2.1 Badge](https://img.shields.io/nexus/r/io.github.mavenplugins/unleash-maven-plugin?server=https://s01.oss.sonatype.org&label=Maven%20Central&queryOpt=:v=3.2.1)](https://central.sonatype.com/artifact/io.github.mavenplugins/unleash-maven-plugin/3.2.1)
+[![3.2.1 Badge](https://img.shields.io/maven-central/v/io.github.mavenplugins/unleash-maven-plugin?label=Maven%20Central&filter=3.2.1)](https://central.sonatype.com/artifact/io.github.mavenplugins/unleash-maven-plugin/3.2.1)
 
 ### Summary
 - Fix vulnerability warnings on dependencies - see **📦 Updates** for details.
@@ -201,7 +201,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [3.2.0]
 <!-- !!! Align version in badge URLs as well !!! -->
-[![3.2.0 Badge](https://img.shields.io/nexus/r/io.github.mavenplugins/unleash-maven-plugin?server=https://s01.oss.sonatype.org&label=Maven%20Central&queryOpt=:v=3.2.0)](https://central.sonatype.com/artifact/io.github.mavenplugins/unleash-maven-plugin/3.2.0)
+[![3.2.0 Badge](https://img.shields.io/maven-central/v/io.github.mavenplugins/unleash-maven-plugin?label=Maven%20Central&filter=3.2.0)](https://central.sonatype.com/artifact/io.github.mavenplugins/unleash-maven-plugin/3.2.0)
 
 ### Summary
 - Preserve leading `0` for increased version parts with the same number of digits if not exceeding the digits required for the new value. - #14<br>
@@ -250,7 +250,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [3.1.0]
 <!-- !!! Align version in badge URLs as well !!! -->
-[![3.1.0 Badge](https://img.shields.io/nexus/r/io.github.mavenplugins/unleash-maven-plugin?server=https://s01.oss.sonatype.org&label=Maven%20Central&queryOpt=:v=3.1.0)](https://central.sonatype.com/artifact/io.github.mavenplugins/unleash-maven-plugin/3.1.0)
+[![3.1.0 Badge](https://img.shields.io/maven-central/v/io.github.mavenplugins/unleash-maven-plugin?label=Maven%20Central&filter=3.1.0)](https://central.sonatype.com/artifact/io.github.mavenplugins/unleash-maven-plugin/3.1.0)
 
 ### Summary
 - Add goals `unleash:nextSnapshotVersion` and `unleash:releaseVersion`
@@ -268,7 +268,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [3.0.3]
 <!-- !!! Align version in badge URLs as well !!! -->
-[![3.0.3 Badge](https://img.shields.io/nexus/r/io.github.mavenplugins/unleash-maven-plugin?server=https://s01.oss.sonatype.org&label=Maven%20Central&queryOpt=:v=3.0.3)](https://central.sonatype.com/artifact/io.github.mavenplugins/unleash-maven-plugin/3.0.3)
+[![3.0.3 Badge](https://img.shields.io/maven-central/v/io.github.mavenplugins/unleash-maven-plugin?label=Maven%20Central&filter=3.0.3)](https://central.sonatype.com/artifact/io.github.mavenplugins/unleash-maven-plugin/3.0.3)
 
 ### Summary
 - 👉 **Reverted** changes from release `3.0.2`
@@ -294,7 +294,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [3.0.2] - DEPRECATED -
 <!-- !!! Align version in badge URLs as well !!! -->
-[![3.0.2 Badge](https://img.shields.io/nexus/r/io.github.mavenplugins/unleash-maven-plugin?server=https://s01.oss.sonatype.org&label=Maven%20Central&queryOpt=:v=3.0.2)](https://central.sonatype.com/artifact/io.github.mavenplugins/unleash-maven-plugin/3.0.2)
+[![3.0.2 Badge](https://img.shields.io/maven-central/v/io.github.mavenplugins/unleash-maven-plugin?label=Maven%20Central&filter=3.0.2)](https://central.sonatype.com/artifact/io.github.mavenplugins/unleash-maven-plugin/3.0.2)
 
 ### Summary
 - DEPRECATED - This release works but changes will be reverted within future releases
@@ -320,7 +320,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [3.0.1]
 <!-- !!! Align version in badge URLs as well !!! -->
-[![3.0.1 Badge](https://img.shields.io/nexus/r/io.github.mavenplugins/unleash-maven-plugin?server=https://s01.oss.sonatype.org&label=Maven%20Central&queryOpt=:v=3.0.1)](https://central.sonatype.com/artifact/io.github.mavenplugins/unleash-maven-plugin/3.0.1)
+[![3.0.1 Badge](https://img.shields.io/maven-central/v/io.github.mavenplugins/unleash-maven-plugin?label=Maven%20Central&filter=3.0.1)](https://central.sonatype.com/artifact/io.github.mavenplugins/unleash-maven-plugin/3.0.1)
 
 ### Summary
 - Fix wrong URLs for **Project URL** and **Source Control** rendered on [Maven Central](https://central.sonatype.com) for this project.
@@ -333,7 +333,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [3.0.0]
 <!-- !!! Align version in badge URLs as well !!! -->
-[![3.0.0 Badge](https://img.shields.io/nexus/r/io.github.mavenplugins/unleash-maven-plugin?server=https://s01.oss.sonatype.org&label=Maven%20Central&queryOpt=:v=3.0.0)](https://central.sonatype.com/artifact/io.github.mavenplugins/unleash-maven-plugin/3.0.0)
+[![3.0.0 Badge](https://img.shields.io/maven-central/v/io.github.mavenplugins/unleash-maven-plugin?label=Maven%20Central&filter=3.0.0)](https://central.sonatype.com/artifact/io.github.mavenplugins/unleash-maven-plugin/3.0.0)
 
 ### Summary
 - Add Workflow action `exitWithRollbackNoError` to abort the workflow  with a rollback of previous actions but with Maven success.<br>
@@ -364,7 +364,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [2.11.0]
 <!-- !!! Align version in badge URLs as well !!! -->
-[![2.11.0 Badge](https://img.shields.io/nexus/r/io.github.mavenplugins/unleash-maven-plugin?server=https://s01.oss.sonatype.org&label=Maven%20Central&queryOpt=:v=2.11.0)](https://central.sonatype.com/artifact/io.github.mavenplugins/unleash-maven-plugin/2.11.0)
+[![2.11.0 Badge](https://img.shields.io/maven-central/v/io.github.mavenplugins/unleash-maven-plugin?label=Maven%20Central&filter=2.11.0)](https://central.sonatype.com/artifact/io.github.mavenplugins/unleash-maven-plugin/2.11.0)
 
 ### Summary
 - Add enhancements provided by RockwellAutomation and hmdebenque
@@ -409,7 +409,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [2.10.0]
 <!-- !!! Align version in badge URLs as well !!! -->
-[![2.10.0 Badge](https://img.shields.io/nexus/r/io.github.mavenplugins/unleash-maven-plugin?server=https://s01.oss.sonatype.org&label=Maven%20Central&queryOpt=:v=2.10.0)](https://central.sonatype.com/artifact/io.github.mavenplugins/unleash-maven-plugin/2.10.0)
+[![2.10.0 Badge](https://img.shields.io/maven-central/v/io.github.mavenplugins/unleash-maven-plugin?label=Maven%20Central&filter=2.10.0)](https://central.sonatype.com/artifact/io.github.mavenplugins/unleash-maven-plugin/2.10.0)
 
 ### Summary
 - Initial release of this artifact with new groupId `io.github.mavenplugins`

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -80,6 +80,23 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 - TBD
 
 
+## [3.3.1]
+<!-- !!! Align version in badge URLs as well !!! -->
+[![3.3.1 Badge](https://img.shields.io/nexus/r/io.github.mavenplugins/unleash-maven-plugin?server=https://s01.oss.sonatype.org&label=Maven%20Central&queryOpt=:v=3.3.1)](https://central.sonatype.com/artifact/io.github.mavenplugins/unleash-maven-plugin/3.3.1)
+
+### Summary
+- Fix issue on Windows OS failing if Maven module relative path names contain upper case characters - #28, #29
+
+### 🐛 Fixes
+- Fix issue on Windows OS failing if Maven module relative path names contain upper case characters - #28, #29
+
+### 🔧 Internal Changes
+- FileToRelativePath.java:
+  - Normalize file path on Windows OS by lower casing drive letter only - #28, #29
+- FileToRelativePathTest.java:
+  - enhance unit test to cover use cases with upper case characters in folder name
+
+
 ## [3.3.0]
 <!-- !!! Align version in badge URLs as well !!! -->
 [![3.3.0 Badge](https://img.shields.io/nexus/r/io.github.mavenplugins/unleash-maven-plugin?server=https://s01.oss.sonatype.org&label=Maven%20Central&queryOpt=:v=3.3.0)](https://central.sonatype.com/artifact/io.github.mavenplugins/unleash-maven-plugin/3.3.0)
@@ -420,7 +437,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 - This is just a dummy placeholder to make the parser of GHCICD/release-notes-from-changelog@v1 happy!
 -->
 
-[Unreleased]: https://github.com/mavenplugins/unleash-maven-plugin/compare/v3.3.0..HEAD
+[Unreleased]: https://github.com/mavenplugins/unleash-maven-plugin/compare/v3.3.1..HEAD
+[3.3.1]: https://github.com/mavenplugins/unleash-maven-plugin/compare/v3.3.0..v3.3.1
 [3.3.0]: https://github.com/mavenplugins/unleash-maven-plugin/compare/v3.2.1..v3.3.0
 [3.2.1]: https://github.com/mavenplugins/unleash-maven-plugin/compare/v3.2.0..v3.2.1
 [3.2.0]: https://github.com/mavenplugins/unleash-maven-plugin/compare/v3.1.0..v3.2.0

--- a/plugin/src/test/java/com/itemis/maven/plugins/unleash/util/functions/FileToRelativePathTest.java
+++ b/plugin/src/test/java/com/itemis/maven/plugins/unleash/util/functions/FileToRelativePathTest.java
@@ -3,6 +3,7 @@ package com.itemis.maven.plugins.unleash.util.functions;
 import java.io.File;
 import java.io.IOException;
 
+import org.apache.commons.lang3.SystemUtils;
 import org.junit.Assert;
 import org.junit.Rule;
 import org.junit.Test;
@@ -18,10 +19,24 @@ public class FileToRelativePathTest {
   @Rule
   public TemporaryFolder tempFolder = new TemporaryFolder();
 
+  public static final String WINDOWS_BASE_PATH_UPPER_DRIVE = "C:\\Temp";
+
+  public static final String WINDOWS_BASE_PATH_LOWER_DRIVE = "c:\\Temp";
+
   @Test
-  @DataProvider({ "test", "1/2/3/xyz.txt" })
+  @DataProvider({ "test", "Test", "1/2/3/xyz.txt" })
   public void testApply(String path) throws IOException {
     File base = this.tempFolder.newFolder();
     Assert.assertEquals(path, new FileToRelativePath(base).apply(new File(base, path)));
   }
+
+  @Test
+  @DataProvider({ "test", "Test", "1/2/3/xyz.txt" })
+  public void testWindowsApply(String path) throws IOException {
+    if (SystemUtils.IS_OS_WINDOWS) {
+      File base = new File(WINDOWS_BASE_PATH_UPPER_DRIVE);
+      Assert.assertEquals(path, new FileToRelativePath(base).apply(new File(WINDOWS_BASE_PATH_LOWER_DRIVE, path)));
+    }
+  }
+
 }

--- a/pom.xml
+++ b/pom.xml
@@ -4,7 +4,7 @@
   <parent>
     <groupId>io.github.mavenplugins</groupId>
     <artifactId>org-parent</artifactId>
-    <version>11</version>
+    <version>12</version>
     <relativePath/>
   </parent>
 

--- a/scm-provider-api/src/main/java/com/itemis/maven/plugins/unleash/scm/utils/FileToRelativePath.java
+++ b/scm-provider-api/src/main/java/com/itemis/maven/plugins/unleash/scm/utils/FileToRelativePath.java
@@ -2,9 +2,7 @@ package com.itemis.maven.plugins.unleash.scm.utils;
 
 import java.io.File;
 import java.net.URI;
-import java.net.URISyntaxException;
 
-import org.apache.commons.lang3.StringUtils;
 import org.apache.commons.lang3.SystemUtils;
 
 import com.google.common.base.Function;
@@ -25,22 +23,24 @@ public class FileToRelativePath implements Function<File, String> {
 
   @Override
   public String apply(File f) {
-    URI workingDirURI = this.workingDir.toURI();
-    URI fileURI = f.toURI();
+    URI workingDirURI = toURIWithNormalizedDriveLetterOnWindows(this.workingDir);
+    URI fileURI = toURIWithNormalizedDriveLetterOnWindows(f);
+    return workingDirURI.relativize(fileURI).toString();
+  }
+
+  private URI toURIWithNormalizedDriveLetterOnWindows(File file) {
     if (SystemUtils.IS_OS_WINDOWS) {
       // On Windows OS deviations in character case of the drive letter may occur
-      // => we have to normalize by lower casing the URI!
-      String lcURIString = StringUtils.EMPTY;
-      try {
-        lcURIString = workingDirURI.toString().toLowerCase();
-        workingDirURI = new URI(lcURIString);
-        lcURIString = fileURI.toString().toLowerCase();
-        fileURI = new URI(lcURIString);
-      } catch (URISyntaxException e) {
-        throw new RuntimeException("Failed to create URI for path: " + lcURIString, e);
+      // => we have to normalize by lower casing the drive letter!
+      // We use file.getAbsoluteFile() to comply with File.toURI inner coding.
+      String absolutePath = file.getAbsoluteFile().getPath();
+      int driveSeparatorIndex = absolutePath.indexOf(':');
+      if (driveSeparatorIndex > 0) {
+        absolutePath = absolutePath.substring(0, driveSeparatorIndex).toLowerCase()
+            + absolutePath.substring(driveSeparatorIndex);
+        return new File(absolutePath).toURI();
       }
     }
-
-    return workingDirURI.relativize(fileURI).toString();
+    return file.toURI();
   }
 }

